File: backend_exanples/calculator.py
# calculator.py - contoh OOP sederhana (CLI)
class Calculator:
    def __init__(self):
        self.current = 0.0

    def add(self, a, b): return a + b
    def sub(self, a, b): return a - b
    def mul(self, a, b): return a * b
    def div(self, a, b):
        if b == 0:
            raise ZeroDivisionError("Pembagian dengan nol")
        return a / b

def run_cli():
    calc = Calculator()
    print("Calculator CLI (ketik 'exit' untuk keluar)")
    while True:
        try:
            expr = input("Masukkan ekspresi (contoh: 2 + 3): ")
            if expr.strip().lower() == 'exit': break
            # simple parse
            parts = expr.split()
            if len(parts) != 3:
                print("Format: <angka> <operator> <angka>")
                continue
            a = float(parts[0]); op = parts[1]; b = float(parts[2])
            if op == '+': print(calc.add(a,b))
            elif op == '-': print(calc.sub(a,b))
            elif op == '*': print(calc.mul(a,b))
            elif op == '/': print(calc.div(a,b))
            else: print("Operator tidak dikenali")
        except Exception as e:
            print("Error:", e)

if __name__ == '__main__':
    run_cli()

File: backend_exanples/calculator.cs
// Calculator.cs - contoh OOP C# (console)
using System;

public class Calculator {
    public double Add(double a,double b) => a + b;
    public double Sub(double a,double b) => a - b;
    public double Mul(double a,double b) => a * b;
    public double Div(double a,double b) {
        if (b == 0) throw new DivideByZeroException();
        return a / b;
    }
}

class Program {
    static void Main() {
        var calc = new Calculator();
        Console.WriteLine("Calculator CLI C# - format: <angka> <operator> <angka>");
        while(true) {
            Console.Write("> ");
            var line = Console.ReadLine();
            if (string.IsNullOrWhiteSpace(line)) continue;
            if (line.Trim().ToLower() == "exit") break;
            var parts = line.Split(' ');
            if (parts.Length != 3) { Console.WriteLine("Format salah"); continue; }
            double a = double.Parse(parts[0]);
            double b = double.Parse(parts[2]);
            var op = parts[1];
            try {
                double r = op switch {
                    "+" => calc.Add(a,b),
                    "-" => calc.Sub(a,b),
                    "*" => calc.Mul(a,b),
                    "/" => calc.Div(a,b),
                    _ => throw new Exception("Operator tidak dikenali")
                };
                Console.WriteLine(r);
            } catch(Exception ex) {
                Console.WriteLine("Error: " + ex.Message);
            }
        }
    }
}